### PR TITLE
Issue 15791 - Cannot get a stored nested struct object from Variant

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: std/variant.d [hunk 1]
```diff
@@ -297,7 +300,9 @@ private:
                     {
                         static if (T.sizeof > 0)
                             assert(target, "target must be non-null");
-                        *zat = *src;
+
+                        //*zat = *src;
```
Comment (atilaneves) [inline std/variant.d]: ?

File: std/variant.d [hunk 2]
```diff
@@ -71,6 +71,9 @@ module std.variant;
 import core.stdc.string, std.conv, std.exception, std.meta, std.traits,
     std.typecons;
 
+private alias Unshared(T) = T;
```
Comment (atilaneves) [inline std/variant.d]: Is Unqual not usable due to constness?

File: std/variant.d [hunk 3]
```diff
@@ -297,7 +300,9 @@ private:
                     {
                         static if (T.sizeof > 0)
                             assert(target, "target must be non-null");
-                        *zat = *src;
+
+                        //*zat = *src;
+                        emplaceRef(*cast(Unshared!T*)zat, *cast(Unshared!A*)src);
```
Comment (atilaneves) [inline std/variant.d]: Why does it have to be unshared?

File: std/variant.d (hunk 1 above)
Comment (9rnsr) [inline std/variant.d]: It was a garbage. Removed.

File: std/variant.d (hunk 2 above)
Comment (9rnsr) [inline std/variant.d]: Yes, `Unqual` was satisfied. I stopped to add a new template `Unshared`.

File: std/variant.d (hunk 3 above)
Comment (9rnsr) [inline std/variant.d]: `Unqual!T` and  `Unqual!A` are enough.

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by atilaneves** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**